src/base/bittorrent/session.cpp

```diff
@@ -4480,54 +4480,63 @@ void Session::startUpTorrents()
         const lt::info_hash_t infoHash = (resumeData.ltAddTorrentParams.ti
                                           ? resumeData.ltAddTorrentParams.ti->info_hashes()
                                           : resumeData.ltAddTorrentParams.info_hashes);
-        if (infoHash.has_v1() && infoHash.has_v2())
+        const bool isHybrid = infoHash.has_v1() && infoHash.has_v2();
+        const auto torrentIDv2 = TorrentID::fromInfoHash(infoHash);
+        const auto torrentIDv1 = TorrentID::fromInfoHash(lt::info_hash_t(infoHash.v1));
+        if (torrentID == torrentIDv2)
         {
-            const auto torrentIDv1 = TorrentID::fromInfoHash(lt::info_hash_t(infoHash.v1));
-            const auto torrentIDv2 = TorrentID::fromInfoHash(infoHash);
-            if (torrentID == torrentIDv2)
+            if (isHybrid && indexedTorrents.contains(torrentIDv1))
             {
-                if (indexedTorrents.contains(torrentIDv1))
+                // if we don't have metadata, try to find it in alternative "resume data"
+                if (!resumeData.ltAddTorrentParams.ti)
                 {
-                    // if we has no metadata trying to find it in alternative "resume data"
-                    if (!resumeData.ltAddTorrentParams.ti)
+                    const std::optional<LoadTorrentParams> loadAltResumeDataResult = startupStorage->load(torrentIDv1);
+                    if (loadAltResumeDataResult)
                     {
-                        const std::optional<LoadTorrentParams> loadAltResumeDataResult = startupStorage->load(torrentIDv1);
-                        if (loadAltResumeDataResult)
-                        {
-                            LoadTorrentParams altResumeData = *loadAltResumeDataResult;
-                            resumeData.ltAddTorrentParams.ti = altResumeData.ltAddTorrentParams.ti;
-                        }
+                        LoadTorrentParams altResumeData = *loadAltResumeDataResult;
```

Comment:
`const`

And it looks like it is making a copy? why not just `loadAltResumeDataResult->ltAddTorrentParams.ti;` on the next line?

Reply:
@Chocobo1 
Changed it in another way.

Reply:
I saw to updated the other if-else branch, but what about this one? is it correct?

Reply:
> I saw to updated the other if-else branch, but what about this one? is it correct? Just double confirming..

Changed.

It was passed during the initial merge, and now I just moved this code from place to place. But for reviewers, it looks like new code (if you don't delve too deeply), so this is another opportunity to improve existing code :+1: